sql/core/src/main/scala/org/apache/spark/sql/internal/SessionState.scala

```diff
@@ -48,8 +48,21 @@ private[sql] class SessionState(sparkSession: SparkSession) {
    * SQL-specific key-value configurations.
    */
   lazy val conf: SQLConf = new SQLConf
-  lazy val hadoopConf: Configuration = {
-    new Configuration(sparkSession.sparkContext.hadoopConfiguration)
+
+  def newHadoopConf(): Configuration = {
+    val hadoopConf = new Configuration(sparkSession.sparkContext.hadoopConfiguration)
+    conf.getAllConfs.foreach { case (k, v) => if (v ne null) hadoopConf.set(k, v) }
```

Comment:
oh, I thought we do not want to override. Then even we directly set sparkContext.hadoopConfiguration, it is still possibly overridden by conf.